mesonbuild/msetup.py

```diff
@@ -219,18 +219,15 @@ def _generate(self, env: environment.Environment) -> None:
         # even to write `T.Callable[[*mlog.TV_Loggable], None]`
         logger_fun = T.cast('T.Callable[[mlog.TV_Loggable, mlog.TV_Loggable], None]',
                             (mlog.log if env.is_cross_build() else mlog.debug))
-        build_machine = intr.builtin['build_machine']
-        host_machine = intr.builtin['host_machine']
-        target_machine = intr.builtin['target_machine']
```

Comment:
Huh... before this PR, was the interpreter copy of each machine guaranteed to be the same one from the original env object we passed in? I *think* so but I'm not sure.

Reply:
that's.... actually a really good question